geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/ManagementControllerAdvice.java

```diff
@@ -91,6 +102,30 @@ public static String removeClassFromJsonText(String json) {
         .replaceAll("\"class\":\"[^\"]*\",", "");
   }
 
+  private static String baseUrl(String requestUrl) {
+    int pos = requestUrl.indexOf(URI_CONTEXT);
+    if (pos >= 0) {
+      return requestUrl.substring(0, pos + URI_CONTEXT.length());
+    } else {
+      pos = requestUrl.indexOf('/', 8);
+      if (pos >= 0) {
+        // for MockMVC tests the URI_CONTEXT will be absent, but tests expect it so fake it
+        return requestUrl.substring(0, pos) + URI_CONTEXT;
+      } else {
+        return null;
+      }
+    }
+  }
+
+  static String qualifyHrefsInJsonText(String json, String requestUrl) {
```

Comment:
if we are manipulating the final links here, maybe we should have all the apis return the links only with the URI_VERSION + uri, and in here add the host and context.

Reply:
it doesn't leave as much to match on, but might be possible to do it that way

Reply:
PR was split so this is no longer relevant